search-service/src/main/kotlin/com/egm/stellio/search/service/EntityPayloadService.kt

```diff
@@ -103,6 +103,99 @@ class EntityPayloadService(
             .bind("specific_access_policy", specificAccessPolicy?.toString())
             .execute()
 
+    @Transactional
+    suspend fun replaceEntity(
+        entityId: URI,
+        payload: String,
+        contexts: List<String>,
+        sub: String? = null
+    ): Either<APIException, Unit> = either {
+        val jsonLdEntity = JsonLdUtils.expandJsonLdEntity(payload, contexts)
+        val ngsiLdEntity = jsonLdEntity.toNgsiLdEntity().bind()
+
+        replaceEntity(entityId, ngsiLdEntity, jsonLdEntity, sub).bind()
+    }
+
+    @Transactional
+    suspend fun replaceEntity(
+        entityId: URI,
+        ngsiLdEntity: NgsiLdEntity,
+        jsonLdEntity: JsonLdEntity,
+        sub: String? = null
+    ): Either<APIException, Unit> = either {
+        val modifiedAt = ngsiLdDateTime()
+        val attributesMetadata = ngsiLdEntity.prepareTemporalAttributes().bind()
+        logger.debug("Replacing entity {}", ngsiLdEntity.id)
+
+        // wait for attributes to be deleted before creating new ones
+        entityAttributeCleanerService.deleteEntityAttributes(entityId).join()
+        replaceEntityPayload(ngsiLdEntity, modifiedAt, jsonLdEntity).bind()
+        temporalEntityAttributeService.createEntityTemporalReferences(
+            ngsiLdEntity,
+            jsonLdEntity,
+            attributesMetadata,
+            modifiedAt,
+            sub
+        ).bind()
+    }
+
+    @Transactional
+    suspend fun replaceEntityPayload(
+        ngsiLdEntity: NgsiLdEntity,
+        modifiedAt: ZonedDateTime,
+        jsonLdEntity: JsonLdEntity
+    ): Either<APIException, Unit> = either {
+        val specificAccessPolicy = ngsiLdEntity.properties.find { it.name == AuthContextModel.AUTH_PROP_SAP }
+            ?.let { getSpecificAccessPolicy(it) }
+            ?.bind()
+        val createdAt = retrieveCreatedAt(ngsiLdEntity.id).bind()
+        replaceEntityPayload(
+            ngsiLdEntity.id,
+            ngsiLdEntity.types,
+            modifiedAt,
+            serializeObject(jsonLdEntity.populateCreatedAtAndModifiedAt(createdAt, modifiedAt).members),
```

Comment:
not sure what you are doing here, but it does not seem to be correct

Reply:
I retrieve the createdAt and I insert in the payload the date of creation and the date of the modification